### PR TITLE
Bump dependencies March 2024

### PR DESCRIPTION
**Pull request by MonsieurNicolas**
@SirTyson was asking me something about sqlite and I realized that we didn't bump it for a while.

This brings core up to date.

Moving to the 3.x Catch2 release seems to be more involved. We may want to do this at some point.

### CODE REVIEW

File: src/util/xdrquery/XDRQueryEval.cpp [hunk 1]
```diff
@@ -472,6 +472,12 @@ FieldList::getFieldNames() const
     return names;
 }
 
+inline std::string
+format_as(const NullField&)
```
Comment (dmkozh) [inline src/util/xdrquery/XDRQueryEval.cpp:476]: What is this change for?

Comment (MonsieurNicolas) [reply src/util/xdrquery/XDRQueryEval.cpp:476]: without this it won't compile with the new version of the fmt library
They fixed a bug in the way `fmt::format` works, and we're supposed to provide custom formatters for custom types

Comment (dmkozh) [reply src/util/xdrquery/XDRQueryEval.cpp:476]: Interesting, do we still need the old formatter defined on line 11?

Comment (MonsieurNicolas) [reply src/util/xdrquery/XDRQueryEval.cpp:463]: yes done, didn't see it

**Comment by MonsieurNicolas** [comment]
@SirTyson please r+ this one when you get a chance (I think you took a dependency on it)

**Comment by anupsdf** [comment]
r+ 2fbd233a324e4426a28d3dc3e8395c0263a0323c

**Comment by anupsdf** [comment]
> @SirTyson please r+ this one when you get a chance (I think you took a dependency on it)

r plussing on behalf of Garand since he doesn't have r plus access yet.

**Pull request merged**